tests/common/plugins/custom_skipif/__init__.py

```diff
@@ -0,0 +1,217 @@
+        inv += ' -i {}'.format(inv_item)
+    return inv
+
+
+def read_skip_file(item, skip_tests_file_path):
+    """
+    Read yaml file with list of test cases which should be skipped
+    :param item: pytest test item
+    :param skip_tests_file_path: path to file where stored list of test cases which should be skipped
+    :return: yaml loaded dictionary
+    """
+    custom_skip_if_dict = 'custom_skip_if_dict'
+    skip_dictionary = item.session.config.cache.get(custom_skip_if_dict, None)
+    if not skip_dictionary:
+        with open(skip_tests_file_path) as skip_data:
+            logger.debug('Reading dynamic skip file: {}'.format(skip_tests_file_path))
+            skip_dictionary = yaml.load(skip_data, Loader=yaml.FullLoader)
+            item.session.config.cache.set(custom_skip_if_dict, skip_dictionary)
+    return skip_dictionary
+
+
+def get_tests_to_be_skipped_path(skip_tests_file='tests_to_be_skipped_conditionally.yaml'):
+    """
+    Get path to file with dynamic skip information
+    :param skip_tests_file: skip test file name
+    :return: full path to skip test file name
+    """
+    custom_skip_folder_path = os.path.dirname(__file__)
+    custom_skip_tests_file_path = os.path.join(custom_skip_folder_path, skip_tests_file)
+    return custom_skip_tests_file_path
+
+
+def update_syspath_for_dynamic_import():
+    """
+    Update sys.path by current folder to have possibility to load python modules dynamically
+    """
+    if os.path.dirname(__file__) not in sys.path:
+        sys.path.append(os.path.dirname(__file__))
+
+
+def prepare_checkers(skip_dict, current_platform):
+    """
+    Import dynamically checker modules and initialize them
+    :param skip_dict: dictionary with skip test case skip conditions
+    :param current_platform: string with current platform, example: x86_64-mlnx_msn2700-r0
+    :return: list with checkers objects
+    """
+    skip_checkers_list = []
+    extra_params = {'current_platform': current_platform}
+    for skip_by in skip_dict:
+        logger.debug('Importing dynamic skip module: {}'.format(skip_by))
+        try:
+            skip_module_obj = eval(
+                '__import__(skip_by).{}({}, {})'.format(skip_by, skip_dict[skip_by], extra_params))
+            skip_checkers_list.append(skip_module_obj)
+        except Exception as err:
+            logger.error('Unable to load dynamically skip object: {}'.format(err))
+    return skip_checkers_list
+
+
+def run_checkers_in_parallel(skip_checkers_list):
+    """
+    Run checkers in parallel and return results
+    :param skip_checkers_list: list with checkers objects
+    :return: dictionary with checkers result
+    """
+    manager = multiprocessing.Manager()
+    skip_dict_result = manager.dict()
+
+    proc_list = list()
+
+    for skip_check in skip_checkers_list:
+        skip_dict_result[skip_check.name] = None
+        proc_list.append(multiprocessing.Process(target=skip_check.is_skip_required, args=(skip_dict_result,)))
```

Comment:
According to https://github.com/google/styleguide/blob/91d6e367e384b0d8aaaf7ce95029514fcdf38651/pyguide.md#218-threading we cannot assume that operations on native python objects such as dicts are atomic. I would suggest adding a lock to avoid bugs.

Reply:
we use in line 140-141 - not default python dict, but multiptocesing.Manager(), as I know it has support for multiprocessing